activerecord/lib/active_record/connection_adapters/mysql/database_statements.rb

```diff
@@ -158,7 +158,7 @@ def exec_stmt_and_free(sql, name, binds, cache_stmt: false)
 
             type_casted_binds = type_casted_binds(binds)
 
-            log(sql, name, binds, type_casted_binds) do
+            log(sql, name, binds, type_casted_binds, nil, background) do
```

Comment:
Seems this won't work if `prepared_statements: true` since `background` isn't positional argument but keyword argument.

```suggestion
            log(sql, name, binds, type_casted_binds, background: background) do
```

Reply:
Indeed, thanks!